manager/integration/tests/test_csi.py

```diff
@@ -195,3 +201,58 @@ def backupstore_test(client, core_api, csi_pv, pvc, pod_make, pod_name, base_ima
     assert resp == test_data
 
     delete_backup(bv, b["name"])
+
+
+@pytest.mark.csi  # NOQA
+def test_csi_block_volume(client, core_api, storage_class, pvc, pod_manifest):  # NOQA
+    pod_name = 'csi-block-volume-test'
+    pvc_name = pod_name + "-pvc"
+    device_path = "/dev/longhorn/longhorn-test-blk"
+
+    storage_class['reclaimPolicy'] = 'Retain'
+    pvc['metadata']['name'] = pvc_name
+    pvc['spec']['volumeMode'] = 'Block'
+    pvc['spec']['storageClassName'] = storage_class['metadata']['name']
+    pvc['spec']['resources'] = {
+        'requests': {
+            'storage': size_to_string(1 * Gi)
+        }
+    }
+    pod_manifest['metadata']['name'] = pod_name
+    pod_manifest['spec']['volumes'] = [{
+        'name': 'longhorn-blk',
+        'persistentVolumeClaim': {
+            'claimName': pvc_name,
+        },
+    }]
+    pod_manifest['spec']['containers'][0]['volumeMounts'] = []
+    pod_manifest['spec']['containers'][0]['volumeDevices'] = [
+        {'name': 'longhorn-blk', 'devicePath': device_path}
+    ]
+
+    create_storage_class(storage_class)
+    create_pvc(pvc)
+    pv_name = wait_and_get_pv_for_pvc(core_api, pvc_name).metadata.name
+    create_and_wait_pod(core_api, pod_manifest)
+
+    test_data = generate_random_data(VOLUME_RWTEST_SIZE)
+    write_pod_block_volume_data(
+        core_api, pod_name, test_data, device_path)
+    md5_sum = get_pod_block_volume_data_md5sum(
```

Comment:
Need to check the raw data once before calculation md5sum. The stored data can be wrong.